src/cursor/cur_table.c

```diff
@@ -804,6 +811,28 @@ __wt_table_range_truncate(WT_CURSOR_TABLE *start, WT_CURSOR_TABLE *stop)
     return (ret);
 }
 
+/*
+ * __wt_cursor_bounds_save --
+ *     Save cursor bounds to restore the state.
+ */
+static inline void
+__wt_cursor_bounds_save(WT_CURSOR *cursor, WT_CURSOR_BOUNDS_STATE *state)
+{
+    WT_ITEM_SET(state->lower_bound, cursor->lower_bound);
+    WT_ITEM_SET(state->upper_bound, cursor->upper_bound);
```

Comment:
I think we need to also save the state of the cursor flags too here. Just in case the bounds function sets a bound flag and we forget to reset it back.

Reply:
[5b20368](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/5b203680c47ab284e59883d194495a9a6703223e)